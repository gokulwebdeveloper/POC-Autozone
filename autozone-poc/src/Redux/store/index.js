import {createStore, applyMiddleware, compose} from 'redux';
import rootReducer from '../reducers';
import rootSaga from '../middlewares/saga';
import createSagaMiddleware from 'redux-saga';

// create the saga middleware
export const sagaMiddleware = createSagaMiddleware();

//list of middlewares
const middlewares = [sagaMiddleware];

//apply middleware on store
const store = createStore(rootReducer, applyMiddleware(...middlewares));

//run saga generator function
sagaMiddleware.run(rootSaga);

export default store;