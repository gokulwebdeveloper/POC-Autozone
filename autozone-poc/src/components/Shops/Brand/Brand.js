import React from 'react'
import { connect } from 'react-redux';
import { filterProducts } from '../../../Redux/actions';
import { changeStyle } from '../ShopHelper.js'

function Brand(props) {
    function handleClick(e) {
        e.preventDefault();
        changeStyle(e);
        props.filterProducts(props.productData, e.currentTarget.dataset.id, e.currentTarget.dataset.title);
    }
    function BrandItem(childProps) {
        return (
            <li >
                <a onClick={handleClick} data-id={childProps.value} data-title='Brands'
                    href="#" >{childProps.value} </a>
            </li>
        );
    }

    return (
        <div className="widget brands mb-50">
            <p className="widget-title2 mb-30">Brands</p>
            <div className="widget-desc">
                <ul>
                    {props.brandList.map((item) => {
                        return <BrandItem value={item} key={item.toString()} />
                    }
                    )}
                </ul>
            </div>
        </div>
    );
}

const mapStateToProps = ({ productData }) => ({
    productData
})
export default connect(mapStateToProps, { filterProducts })(Brand);

