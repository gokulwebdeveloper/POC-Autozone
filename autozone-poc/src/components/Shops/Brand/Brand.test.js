import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";
import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Brand from './Brand';
import store from '../../../Redux/store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });


describe('Brand testing', () => {
  var brandList = ["Asos",
    "Mango",
    "River Island",
    "Topshop",
    "Zara"];

    beforeEach(() => {
      
    });

  let wrapper = mount(<Provider store={store}><Brand brandList={brandList} /></Provider>);

  it("SnapShot Match for Brand component", () => {
    const page = renderer.create(<Provider store={store}><Brand brandList={brandList} /></Provider>).toJSON();
    expect(page).toMatchSnapshot();
  });

  it("displays brand list", () => {
    expect(wrapper.find("BrandItem").first().text()).toBe("Asos ");
    expect(wrapper.find("BrandItem").at(1).text()).toBe("Mango ");
    expect(wrapper.find("BrandItem").at(2).text()).toBe("River Island ");
    expect(wrapper.find("BrandItem").at(3).text()).toBe("Topshop ");
    expect(wrapper.find("BrandItem").at(4).text()).toBe("Zara ");
  });

  it("displays brand title", () => {
    expect(wrapper.find("p").first().text()).toBe("Brands");

  });

  it("li value in BrandItem", () => {
    expect(wrapper.find("li").first().text()).toBe("Asos ");
    expect(wrapper.find("li").at(1).text()).toBe("Mango ");
    expect(wrapper.find("li").at(2).text()).toBe("River Island ");
    expect(wrapper.find("li").at(3).text()).toBe("Topshop ");
    expect(wrapper.find("li").at(4).text()).toBe("Zara ");
  });

  it("BrandItem click action", () => {
    wrapper.find("li").first().simulate('click');
    console.log("store data : " + JSON.stringify(store.getState()) ) 
  });

})