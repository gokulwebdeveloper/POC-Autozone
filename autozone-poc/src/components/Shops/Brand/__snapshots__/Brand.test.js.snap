// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand testing SnapShot Match for Brand component 1`] = `
<div
  className="widget brands mb-50"
>
  <p
    className="widget-title2 mb-30"
  >
    Brands
  </p>
  <div
    className="widget-desc"
  >
    <ul>
      <li>
        <a
          href="#"
        >
          Asos
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Mango
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          River Island
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Topshop
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Zara
        </a>
      </li>
    </ul>
  </div>
</div>
`;
